sdk/inc/azure/core/az_json.h

```diff
@@ -422,119 +369,127 @@ AZ_NODISCARD az_result az_json_builder_append_end_array(az_json_builder* json_bu
+ * @brief Initializes an #az_json_parser to parse the JSON payload contained within the provided
+ * buffer.
  *
- * @param json_parser A pointer to an az_json_parser instance containing the JSON to parse.
- * @param out_token_member A pointer to an az_json_token_member containing the next parsed JSON
- * token member.
- * @return AZ_OK if the token was parsed successfully.<br>
- *         AZ_ERROR_EOF when the end of the JSON document is reached.<br>
- *         AZ_ERROR_PARSER_UNEXPECTED_CHAR when an invalid character is detected.<br>
- *         AZ_ERROR_ITEM_NOT_FOUND when no more items are found.
+ * @param[out] json_parser A pointer to an #az_json_parser instance to initialize.
+ * @param[in] json_buffer An #az_span over the byte buffer containing the JSON text to parse.
+ * @param[in] options __[nullable]__ A reference to an #az_json_parser_options
+ * structure which defines custom behavior of the #az_json_parser. If `NULL` is passed, the parser
+ * will use the default options (i.e. #az_json_parser_options_default()).
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if the az_json_parser is initialized successfully
+ *         - #AZ_ERROR_EOF if the provided json buffer is empty
  */
-AZ_NODISCARD az_result az_json_parser_parse_token_member(
+AZ_NODISCARD AZ_INLINE az_result az_json_parser_init(
     az_json_parser* json_parser,
-    az_json_token_member* out_token_member);
+    az_span json_buffer,
+    az_json_parser_options const* options)
+{
+  // TODO: Should this be a precondition? If so, there is an inlining issue here.
+  // An empty JSON payload is invalid.
+  if (az_span_size(json_buffer) < 1)
+  {
+    return AZ_ERROR_EOF;
+  }
+
+  *json_parser = (az_json_parser){
+    .token = (az_json_token){
+      .kind = AZ_JSON_TOKEN_NONE,
+      .slice = AZ_SPAN_NULL,
+      ._internal = {
+        .string_has_escaped_chars = false,
+      },
+    },
+    ._internal = {
+      .json_buffer = json_buffer,
+      .bytes_consumed = 0,
+      .is_complex_json = false,
+      .bit_stack = { 0 },
+      .options = options == NULL ? az_json_parser_options_default() : *options,
+    },
+  };
+  return AZ_OK;
+}
 
 /**
- * @brief az_json_parser_parse_array_item returns the next array item in the JSON document.
+ * @brief Reads the next token in the JSON text and updates the parser state.
+ *
+ * @param json_parser A pointer to an #az_json_parser instance containing the JSON to parse.
  *
- * @param json_parser A pointer to an az_json_parser instance containing the JSON to parse.
- * @param out_token A pointer to an az_json_token containing the next parsed JSON array item.
  * @return AZ_OK if the token was parsed successfully.<br>
  *         AZ_ERROR_EOF when the end of the JSON document is reached.<br>
  *         AZ_ERROR_PARSER_UNEXPECTED_CHAR when an invalid character is detected.<br>
  *         AZ_ERROR_ITEM_NOT_FOUND when no more items are found.
```

Comment:
We don't ever return `AZ_ERROR_ITEM_NOT_FOUND` anymore. I am going to remove this from the docs unless someone can come up with a case I might have missed.